program-runtime/src/invoke_context.rs

```diff
@@ -37,7 +37,20 @@ use {
     },
 };
 
-pub type ProcessInstructionWithContext = fn(&mut InvokeContext) -> Result<(), InstructionError>;
+#[macro_export]
+macro_rules! declare_process_instruction {
```

Comment:
Please add a comment explaining what this is for. In the context of this PR is
clear, but if you only look at the call sites they look pretty weird so people
will wonder what's going on